sql/collations.go

```diff
@@ -972,3 +973,47 @@ type TypeWithCollation interface {
 	// whether to include the character set and/or collation information.
 	StringWithTableCollation(tableCollation CollationID) string
 }
+
+// ConvertCollationID converts numeric collation IDs to their string names.
+func ConvertCollationID(val any) (string, error) {
```

Comment:
There's a `collations_test.go` file that would be a good place to add a quick unit test for this logic